compute_tools/src/lsn_lease.rs

```diff
@@ -0,0 +1,187 @@
+use anyhow::bail;
+use anyhow::Context;
+use anyhow::Result;
+use postgres::{NoTls, SimpleQueryMessage};
+use std::time::SystemTime;
+use std::{str::FromStr, sync::Arc, thread, time::Duration};
+use utils::id::TenantId;
+use utils::id::TimelineId;
+
+use compute_api::spec::ComputeMode;
+use tracing::{info, warn};
+use utils::{
+    lsn::Lsn,
+    shard::{ShardCount, ShardNumber, TenantShardId},
+};
+
+use crate::compute::{ComputeNode, ComputeState};
+
+/// Spawns a background thread to periodically renew LSN leases for static compute.
+/// Do nothing if the compute is not in static mode.
+pub fn launch_lsn_lease_bg_task_for_static(compute: &Arc<ComputeNode>) {
+    let lsn = {
+        let state = compute.state.lock().unwrap();
+        let spec = state.pspec.as_ref().expect("spec must be set");
+        match spec.spec.mode {
+            ComputeMode::Static(lsn) => lsn,
+            _ => return,
```

Comment:
Started writing that to propose you to add a compute feature flag and use it here
https://github.com/neondatabase/neon/blob/6949b45e1795816507f5025a474e15d718e07456/libs/compute_api/src/spec.rs#L34C23-L34C37

But then realized that currently, the control plane doesn't start static endpoints at all, and there is a separate feature flag to start using them `static_ephemeral_endpoints`
https://github.com/neondatabase/cloud/blob/cfdadee070fa3503048ef7242da50f26bed1c4b0/goapp/internal/dto/account_settings.go#L146

And it means when we enable it -- your code will kick in, but when it's disabled, it shouldn't affect any other workload

So leaving it just for the context, maybe someone didn't know about these flags :)

Reply:
Thanks for the information. Good to know these! Maybe we should just keep using the control plane flags? I believe we need the leases for the static compute to work properly with pageserver GC.

Reply:
Yeah, that was the point -- looks like the current flag is enough